// mySQLconnect/mySQLconnect.js

//mySQLconnect Module

exports.queryInsert = queryInsert;
exports.querySelect = querySelect;
exports.connection = connection;

// Create a connection to MySql Server and Database
var connection = require('mysql').createConnection({
  host : 'localhost',
  user : 'admin',
  password : '1234'
});

var TB_CHAT = 'simpoze.cht_chat';
var SELECT_Q = "SELECT * from "+TB_CHAT;
var INSERT_Q = "INSERT INTO " +TB_CHAT+ " (username, text) VALUES ";


function querySelect(socket){
	var query = SELECT_Q;
	if (!query){ //ERROR in query
			console.log("Error in Query! ");
			return; 
		}

	console.log(query);
    connection.query(query, function(err, rows){
        if(err != null) { 
            socket.emit("DB", "Error in Query: " + err);
        }else{// Shows the results:
			for (var i=0; i<rows.length; i++){
				 console.log(rows[i]);
				 socket.emit("DBnews",rows[i]);
			}
        }
    });
}

function queryInsert(socket, data){
	var query = INSERT_Q;
	if (!query){ //ERROR IN QUERY
			console.log("Error in Query! ");
			return; 
		}
	query += "('"+ data.username +"','" + data.message +"')";
	console.log(query);
    connection.query(query, function(err, rows){
        if(err != null) {
            socket.emit("DBnews", "Error in Query: " + err);
        }
    });
}
